lib/utils/os.ts

```diff
@@ -0,0 +1,439 @@
+/**
+ * @license
+ * Copyright 2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This module is a copy-and-paste (with minor amendments) from:
+ * https://github.com/balena-io/balena-ui/blob/v2.497.9/src/services/os.ts
```

Comment:
This PR is still a draft, for discussion. This whole file was copied from:
https://github.com/balena-io/balena-ui/blob/v2.497.9/src/services/os.ts

As kindly pointed out by @JSReds in [this flowdock thread](https://www.flowdock.com/app/rulemotion/resin-devices/threads/iHHXQlkB5Dt-m2_EBzO3WlDBn1G).

I would be able to trim it down a bit (not all exported functions are used by the `balena os versions` command - in fact only `getAllOsVersions` is used - but some other function may still be useful for other CLI commands like `balena os download` - pending investigation).

As this file proved to be useful for both the web dashboard and the CLI, it should probably be part of the balena SDK, no? Thoughts?

Reply:
@pdcastro it indeed should be in the SDK, and I opened a draft PR [here](https://github.com/balena-io/balena-sdk/pull/876). The problem is that we currently don't have a good way to preload hostapps on open balena and other environments, so I switched to working on that before I update the OS methods, so we don't have to have `os.ts` and `newOs.ts` distinction. If this is urgent, however, I can push on that next week and get it merged in the SDK.